Docfx v3 migration

VSC-Service-Account opened the pull request:
Migration PR to turn docfx build engine to v3

Commit:
migration-content-fix: fixed redirections in yaml header

Commit:
migration-content-fix: fixing invalid redirection with id

Commit:
Update config to enable docfx v3

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=16209) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=16209) before we can accept your contribution.<br/><br/>:x: VSC-Service-Account [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=16209)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=16209) it.</sub>

928PJY committed:
Use `title` instead of `titleOverwriteH1`

`titleOverwriteH1` is not supported in v3 now, this change will not bring any impact to your current published page.

BillWagner commented:
@mairaw Do you know why the redirections are changed here? We were using those file based redirections intentionally to remind our team to update them when there were new releases.

If this change is mandatory, there should be a few additional changes:
1. The redirection file should be sorted alphabetically per our process.
1. The TOCs for those intentional file-based redirections should have comments reminding our future selves to update the redirects when new defaults are added.

Thraka commented:
Shoooot, I have a PR open that adds 30-40 redirects. Now I'll have to wait for this! 😠

mairaw committed:
fix build warnings

mairaw committed:
update url

mairaw commented:
It'll be easier to see the changes to the redirection file if you hide the white space changes. I've removed the redirect_document_id true if the article moved to a different location.

mairaw committed:
remove redirect true from non repo articles

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -1,3811 +1,3806 @@
-            "redirect_url": "/dotnet/csharp/language-reference/operators/namespace-alias-qualifier"
-        },
-        {
-            "source_path": "docs/
```
@Thraka I had to change this from /dotnet/desktop-wpf/fundamentals/how-to-create-apply-template to /dotnet/desktop-wpf/themes/how-to-create-apply-template. But it seems odd that most of the content is in the fundamentals folder except this one article.

Thraka replied:
I'm going to be moving more stuff around.

mairaw committed:
sort/fix/updateeachrelease

mairaw commented:
@BillWagner @Thraka I think this is ready to go. Please take a look at the redirection file and see if you find any oddities there.

Thraka commented on .openpublishing.redirection.json:
```diff
@@ -1,3811 +1,3798 @@
-            "redirect_url": "/dotnet/csharp/language-reference/operators/namespace-alias-qualifier"
-        },
-        {
-            "source_path": "docs/
```
@mairaw 
Don't these need to be `docs/desktop-wpf/data/data-binding-overview` ??

mairaw replied:
Good point! I think the tool just copied what was in the metadata.

Thraka commented on .openpublishing.redirection.json:
```diff
@@ -1,3811 +1,3798 @@
-            "redirect_url": "/dotnet/csharp/language-reference/operators/namespace-alias-qualifier"
-        },
-        {
-            "source_path": "docs/
```
Shouldn't we keep this doc id redirect?

```suggestion
      "redirect_document_id": true
      "redirect_url": "/dotnet/desktop-wpf/themes/how-to-create-apply-template"
```

mairaw replied:
But now you've already published the new article without the proper redirect. So transferring the BI data now could mess things up, I think.

mairaw replied:
Because when you do that in sync, the new article gets the id of the old one when published. The new one has been live for a few months now so it has its own BI data now, if that makes sense.

BillWagner reviewed (approved):
Pending Andy's questions, this LGTM.

fenxu commented:
Thank you all @mairaw  @BillWagner @Thraka

Thraka commented on .openpublishing.redirection.json:
```diff
@@ -1,3811 +1,3798 @@
-            "redirect_url": "/dotnet/csharp/language-reference/operators/namespace-alias-qualifier"
-        },
-        {
-            "source_path": "docs/
```
Is this valid in the redirect file?

mairaw replied:
yes, I explained it in the e-mail that the redirect file can have custom fields.

mairaw committed:
fix redirect paths

mairaw committed:
fix bad redirect
